CHANGELOG.md

```diff
@@ -1,5 +1,49 @@
+
+* Bug fixes:
+  * Updated the ACNS Image tag for fixing the [bug](https://github.com/Azure/AKS/issues/4525) that causes cilium pods to creash in ACNS enabled AKS clusters
+  * Bug fix to address the issue where the OSDiskSize validator throws an error if the existing agent pool does not have a default value set
+  
+* Behavior change:
+  * Abandoned cluster will be deallocated with status `Failed(Deallocated)` instead of `Succeeded (Stopped)`.
+  * PDB drain errors will now include additional PDB debug message and appropriate original error instead of generic "API call to Kubernetes API Server failed" error message.
+  * Updated [Azure NPM version to v1.5.36](https://github.com/Azure/azure-container-networking/releases/tag/v1.5.36) to address race condition in Azure NPM Linux which can occur when editing/deleting a NetworkPolicy with "enough" rules. The race can result in unexpected connectivity for traffic to/from Pods on the impacted Node. NPM will now auto-restart to mitigate the issue ~15 seconds after if it enters a broken state caused by the race.
+  * In k8s 1.31, we will default to parallel image pulling, by setting the kubelet flag --serialize-image-pulls = false(default value is true)
+  * Lowering Linux Azure NPM's CPU request from 250m to 50m. This addresses [Github Issue 2792](https://github.com/Azure/AKS/issues/2792)
+  * Konnectivity + Private Keyvault will not be allowed starting from kubernetes version 1.31
+
+* Component updates:
+  * All revisions of [Azure Service Mesh](https://learn.microsoft.com/azure/aks/servicemesh-about) use zipkin as the default tracer config.
+  * [Cost-analysis-agent](https://learn.microsoft.com/azure/aks/cost-analysis) image upgraded from v0.0.16 to v0.0.17.
+  * Updated windows image to [retina-agent]() from kappie-agent. Updated [retina](https://learn.microsoft.com/azure/aks/network-observability-overview) linux to v0.0.15.
+  * Updated [ip-masq-agent](https://github.com/Azure/ip-masq-agent-v2/compare/v0.1.11...v0.1.13) to v0.1.13 to address [CVEs](). 
+  * Updated [aci connector addon](https://learn.microsoft.com/azure/aks/virtual-nodes) to v1.6.2 and init-validation to v0.3.0.
+  * Updated Azure Disk CSI driver version to [v1.29.9](https://github.com/kubernetes-sigs/azuredisk-csi-driver/releases/tag/v1.29.9)  on AKS 1.28, 1.29, v1.30.4  on AKS 1.30.
+  * Updated Azure File CSI driver to v1.29.8 on AKS 1.28.
+  * Updated [tigera operator to v1.30.11](https://github.com/tigera/operator/releases/tag/v1.30.11) and [calico to v3.26.5](https://github.com/projectcalico/calico/releases/tag/v3.26.5) for versions running on k8s 1.29 and 1.30 to address CVE patches.
+  * Updated Extension Manager to [v1.18.3](https://dev.azure.com/msazure/CloudNativeCompute/_git/aks-rp/pullrequest/10782116).
```

Comment:
URL not public. Not sure if this is a public change either.

Reply:
Do we have links for all the update in docs or GH?